src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -192,25 +194,32 @@ public String  MakeGUID() {
     public void BeginConnectPeer(String toPeerId, ConnectStatusCallback connectStatusCallback) {
         ConnectStatusCallback tmpCallback = connectStatusCallback;
 
+        // needed since mBtToRequestSocket can be disconnected from other thread
```

Comment:
@DrJukka What happens if BeginConnectPeer is called twice in parallel? This is perfectly reasonable if we are trying to connect to two different peers. I'm fairly confident that making the method synchronized is not the right solution. This, for example, won't solve what happens if someone tries to simultaneously connect and disconnect since synchronized methods apply to repeated calls to the method, not the object. I"m not worried about race conditions as much as I'm worried about us trying to connect to the same peer twice. Will Bluetooth's layer stop that? Or can you open two Bluetooth client sockets to the same device?